main.ts

```diff
@@ -0,0 +1,150 @@
+let gameBoard: number[][] = [];
+let gameSize: number = parseInt(prompt("Specify size of Board"));
+let winCon: number = parseInt(prompt("Specify how many in a row to win."));
+while (gameSize<winCon || (gameSize === NaN || winCon === NaN)) {
+  gameSize = parseInt(prompt("Specify size of Board"));
+  winCon = parseInt(prompt("Specify how many in a row to win."));    
+}
+var boxClicked = function() {
+  let x: number = parseInt(this.getAttribute("data-x"));
+  let y: number = parseInt(this.getAttribute("data-y"));
+  let state: number = parseInt(this.getAttribute("data-state"));
+  console.log("User clicked on box " + x + " " + y);
+  let currentText: string = this.childNodes[0].innerText;
+  if (!(currentText === "X" || currentText === "O")) {
+    let oneOrTwo: number = circleCross();
+    let letter: string;
+    if (oneOrTwo === 1) {
+      //return an X
+      letter = "X";
+    } else {
+      //return an O
+      letter = "O";
+    }
+    this.childNodes[0].innerText = letter;
+    //assign that oneOrTwo to the array representation
+    gameBoard[x][y] = oneOrTwo;
+    checkWin();
+  }
+};
+
+function setupBoard() {
```

Comment:
Good function to dynamically setup the board, allows the game to be scaled to larger sizes